### PR TITLE
bpf: ignore errno 2 when deleting non-existing entry

### PR DESCRIPTION
**Pull request by jxiaobin**
```release-note
Fix bug when deleting an non-existing entry
```

Note: errno 2 is "no such file or directory".

Previously if an entry doesn't exist in underlying map, an error will be returned
directly and the cache entry will be left there. This fix logs such event and
ignores the error so that cache deletion will go through.

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

### CODE REVIEW

File: pkg/bpf/map_linux.go [hunk 1]
```diff
@@ -981,7 +981,12 @@ func (m *Map) Delete(key MapKey) error {
 		metrics.BPFMapOps.WithLabelValues(m.commonName(), metricOpDelete, metrics.Errno2Outcome(errno)).Inc()
 	}
 	if errno != 0 {
-		err = fmt.Errorf("unable to delete element %s from map %s: %w", key, m.name, errno)
+		if errno == 2 {
```
Comment (christarazi) [inline pkg/bpf/map_linux.go:984]:
```suggestion
		if errno == unix.ENOENT {
```

File: pkg/bpf/map_linux.go [hunk 2]
```diff
@@ -981,7 +981,12 @@ func (m *Map) Delete(key MapKey) error {
...
+			// ignore errno 2 "no such file or directory"
+			m.scopedLogger().WithField("element", key).WithField("map", m.name).Warn("element not found")
```
Comment (christarazi) [inline pkg/bpf/map_linux.go:986]:
```suggestion
			m.scopedLogger().WithFields(logrus.LogFields{
			    "key": key,
			    "map": m.name,
			}).Warn("element not found")
```

It's worth seeing if we already having `logfields` for each field in the above entry, i.e. `logfields.MapKey`, `logfield.MapName`, etc.

**Review by christarazi — COMMENTED** [review]
Overall looks good, just minor suggestions.

Also, could you rephrase the release-note to be more specific about which entry?

**Review by brb — CHANGES_REQUESTED** [review]
Thanks.

I'd rather keep the error, as there might be the API consumers which are interested in ENOENT when removing an entry. Instead, let's better figure out why `pkg/service` is trying to delete a non-existing entry. It should have an up to date view of the relevant BPF maps content.

**Comment by pchaigno** [comment]
@jxiaobin I've converted the pull request to draft. If you're still working on this, don't hesitate to move out of draft once it's ready for a new round of reviews. (Although, looking at Martynas's review, it seems we should debug the error rather than hide it.)
